examples/flax/language-modeling/run_bart_dlm_flax.py

```diff
@@ -799,19 +799,25 @@ def loss_fn(params):
             loss = optax.softmax_cross_entropy(logits, onehot(labels, logits.shape[-1])) * label_mask
 
             # take average
-            loss = loss.sum() / label_mask.sum()
+            loss = loss.sum()
+            num_labels = label_mask.sum()
 
-            return loss
+            return loss, num_labels
 
-        grad_fn = jax.value_and_grad(loss_fn)
-        loss, grad = grad_fn(state.params)
-        grad = jax.lax.pmean(grad, "batch")
-        new_state = state.apply_gradients(grads=grad)
+        grad_fn = jax.value_and_grad(loss_fn, has_aux=True)
+        (loss, num_labels), grad = grad_fn(state.params)
+        num_labels = jax.lax.psum(num_labels, "batch")
 
-        metrics = jax.lax.pmean(
-            {"loss": loss, "learning_rate": linear_decay_lr_schedule_fn(state.step)}, axis_name="batch"
-        )
+        # true loss = total loss / total samples
+        loss = jax.lax.psum(loss, "batch")
+        loss = jax.tree_map(lambda x: x / num_labels, loss)
```

Comment:
(nit) `jax.tree_map` is deprecated and will be replaced by `jax.tree_util.tree_map` in future versions of JAX
```suggestion
        loss = jax.tree_util.tree_map(lambda x: x / num_labels, loss)
```

Reply:
Thanks! I've replaced all.